include/boost/geometry/algorithms/detail/overlay/discard_duplicate_turns.hpp

```diff
@@ -137,6 +137,13 @@ inline void discard_duplicate_start_turns(Turns& turns,
                 {
                     for (std::size_t const& i : it->second)
                     {
+                        if (turns[i].cluster_id != turn.cluster_id)
+                        {
+                            // The turns are not part of the same cluster,
+                            // or one is clustered and the other is not.
+                            // This is not corresponding.
+                            continue;
```

Comment:
This was earlier not possible. But now that clusters are detected before, the start turn and its corresponding turn should be located on the same cluster.

Reply:
We might be able to delete the `corresponding_turn` functions because of this.
Will check that, but if possible, it will be a different PR